python/py-apache-beam/Portfile

```diff
@@ -3,9 +3,20 @@
 PortSystem          1.0
 PortGroup           python 1.0
 
+# Set known_fail for macOS versions > 13 and < 10.10; can be removed if issue is fixed
+platform darwin {
+    if {${os.major} > 22 || ${os.major} < 14} {
+        known_fail  yes
+        pre-fetch {
+            ui_error \
+            "${name} is known to fail on macOS versions earlier than 10.10 and later than 13."
+            return -code error "incompatible OS version"
+        }
+    }
+}
+
```

Comment:
For cases where a port is only available on a range of macOS releases, you can specify that via a special syntax for `platforms`.

Here are a few examples:

* This case:
    `platforms     {darwin >= 14 <= 22}`

* Unbounded ranges:
    `platforms     {darwin > 20}`
    `platforms     {darwin < 15}`

You can replace the entire block with a one-line `platforms` statement.